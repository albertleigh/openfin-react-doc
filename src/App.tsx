import * as React from 'react';
import cx from 'classnames';
import { connect } from 'react-redux';
import logo from './logo.svg';
import { IRootState }from './reduxs';
import './App.css';

interface IProps {
  loading:boolean,
  [key:number]:any,
  [key:string]:any,
}

interface IState {
  [key:number]:any,
  [key:string]:any,
}

class App extends React.Component<IProps,IState> {
  render() {

    const { loading } = this.props;

    return (
      <div className="App">
        <header className="App-header">
          <p>
              {loading?'openfin-react-doc Loading...':'openfin-react-doc Loaded~'}
          </p>
          <img src={logo} className={cx(
              'animated','slow','infinite', 'App-logo',
              {
                  flash:loading,
                  ["App-logo-spin"]:!loading,
              }
          )} alt="logo" />
          <p>
            Edit <code>src/App.tsx</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default connect(
    (state:IRootState)=>({
        loading:state.application.loading,
    }),
    dispatch => ({
        actions:{

        }
    })

)(App);
