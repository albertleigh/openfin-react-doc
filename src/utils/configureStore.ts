import { applyMiddleware, createStore, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootReducer from '../reduxs';
import rootSaga from '../reduxs/sagas';

declare const window:any;

export default ()=>{

    const sagaMiddleware = createSagaMiddleware();
    const devtools = window.devToolsExtension?window.devToolsExtension():(f:any):any => (f);

    const middleware = compose(
        applyMiddleware(
            sagaMiddleware,
        ),
        devtools
    );

    const store = createStore(
        rootReducer,
        middleware,
    );

    sagaMiddleware.run(rootSaga as any);

    return store;

}
