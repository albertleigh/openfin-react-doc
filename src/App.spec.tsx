import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import App from './App';


const mockStore = configureStore();

describe('App Entry',()=>{
    it('render while loading',()=>{
        const initialState = {
          application:{
            loading:true,
          }
        };
        const store = mockStore(initialState);
        const wrapper = shallow(<App store={store}/>);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });

    it('render while loaded',()=>{
        const initialState = {
            application:{
                loading:false,
            }
        };
        const store = mockStore(initialState);
        const wrapper = shallow(<App store={store}/>);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });
});