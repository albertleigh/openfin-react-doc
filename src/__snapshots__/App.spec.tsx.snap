// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Entry render while loaded 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      openfin-react-doc Loaded~
    </p>
    <img
      alt="logo"
      className="animated slow infinite App-logo App-logo-spin"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;

exports[`App Entry render while loading 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      openfin-react-doc Loading...
    </p>
    <img
      alt="logo"
      className="animated slow infinite App-logo flash"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
